Repost of PR 34

Darsstar opened the pull request:
https://github.com/SpoutDev/Vanilla/pull/34

Had a mindfart or something...

Zidane commented:
The formatting seems a little screwy on this PR (some classes being wiped and redone is an indicator).

Also, how about GenericFullContainerItem as a name instead of what you have? As it is a block in the world, item when a player has it and sometimes an entity (if it moves).

Darsstar committed:
Placeable lava and water when using buckets and get an empty bucket in your inventory.
Scoop up lava and water using buckets and get a water/lava bucket in your inventory.
Added Override annotations.
Refactored the gamemode command.
Prevented a whole lot of NPEs originating from MovingEntity.
Signed-off-by:Darsstar <darsstar@gmail.com>
